benchmarking/benchmark.py

```diff
@@ -0,0 +1,380 @@
+
+    # testing for different sizes of input datasets
+    result = []
+    for data_fraction in np.linspace(0, 1, N_SPLITS + 1)[1:]:
+        n = int(n_molecules * data_fraction)
+        subset = X[:n]
+        times = [None for _ in range(N_REPEATS)]
+        # testing several times to get average computation time
+        for i in range(N_REPEATS):
+            start = time()
+            X_transformed = emf_transformer.transform(subset)
+            end = time()
+            times[i] = end - start
+        result.append(np.mean(times))
+    return np.array(result)
+
+
+def get_generator_times_rdkit(
+    X: pd.DataFrame, generator: object, count: bool, sparse: bool
+):
+    n_molecules = X.shape[0]
+    if count:
+        fp_function = lambda x: generator.GetCountFingerprint(
+            MolFromSmiles(x)
+        ).ToList()
+    else:
+        fp_function = lambda x: generator.GetFingerprint(MolFromSmiles(x))
+
+    # testing for different sizes of input datasets
+    result = []
+    for data_fraction in np.linspace(0, 1, N_SPLITS + 1)[1:]:
+        n = int(n_molecules * data_fraction)
+        subset = X[:n]
+        times = [None for _ in range(N_REPEATS)]
+        # testing several times to get average computation time
+        for i in range(N_REPEATS):
+            start = time()
+            if sparse:
+                X_transformed = csr_array([fp_function(x) for x in subset])
+            else:
+                X_transformed = np.array([fp_function(x) for x in subset])
+            end = time()
+            times[i] = end - start
+        result.append(np.mean(times))
+    return np.array(result)
+
+
+def get_times_rdkit(
+    X: pd.DataFrame, func: Callable, sparse: bool = False, **kwargs
+):
+    n_molecules = X.shape[0]
+    # testing for different sizes of input datasets
+    result = []
+    for data_fraction in np.linspace(0, 1, N_SPLITS + 1)[1:]:
+        n = int(n_molecules * data_fraction)
+        subset = X[:n]
+        times = [None for _ in range(N_REPEATS)]
+        # testing several times to get average computation time
+        for i in range(N_REPEATS):
+            start = time()
+            if sparse:
+                X_transformed = csr_array(
+                    [func(MolFromSmiles(x), **kwargs) for x in subset]
+                )
+            else:
+                X_transformed = np.array(
+                    [func(MolFromSmiles(x), **kwargs) for x in subset]
+                )
+            end = time()
+            times[i] = end - start
+        result.append(np.mean(times))
+    return np.array(result)
+
+
+def plot_results(
+    n_molecules: int,
+    y_emf: List,
+    y_rdkit: List,
+    title: str = "",
+    sparse: bool = None,
+    count: bool = None,
+    save: bool = True,
+):
+    if sparse is not None:
+        if sparse:
+            title += " sparse"
+
+    if count is not None:
+        if count:
+            title += " count"
+        else:
+            title += " bit"
```

Comment:
According to typing, count can't be `None`. So `if count` would suffice

Reply:
Changed to:
```
    if count:
        title += " count"
    elif count is not None:
        title += " bit"
```
And added typing. We don't want "bit" in the name of fingerprints that don't have a "count" parameter